File: src/toNumber.js

// usage: node toNumber.js string


if(process) {
  console.log('Node version:', process.version, 'V8', process.versions.v8);
}

let st = Date.now();
const x=process.argv[2];


const count=100000000;
for(let i=0; i<count; i++) {
   Number.isNaN(Number(x))
}

console.log(`Number('${x}') rate: `, Math.round(count/(Date.now()-st)), 'Kps')

st = Date.now();
for(let i=0; i<count; i++) {
   Number.isNaN(+x)
}
console.log(`(+'${x}') rate: `, Math.round(count/(Date.now()-st)), 'Kps')


st = Date.now();
for(let i=0; i<count; i++) {
   Number.isNaN(parseFloat(x))
}
console.log(`parseFloat('${x}') rate: `, Math.round(count/(Date.now()-st)), 'Kps')

File: src/toNumber.c
#include "stdio.h"
#include "stdlib.h"

#include <sys/time.h>
#include <sys/resource.h>

double now()
{
    struct timeval t;
    struct timezone tzp;
    gettimeofday(&t, &tzp);
    return (t.tv_sec + t.tv_usec*1e-6)*1000;
}


int main(int argc, char* argv[]) {
   if(argc < 2) {
     printf("usage: %s <string>\n", argv[0]);
     exit(1);
   }
   const char* x = argv[1];
   double k = 0;
   size_t count = 100000000;
   double start = now();
   for(size_t i=0; i<count; i++) {
      k = atof(x);
   }
   
   printf("rate, x=%s: %.0f Kps\n", x, (count/(now() - start)));

}


